File: ai25btech11007/MatGeo/MatGeo_10/Codes/main.py
import numpy as np

def check_coplanarity(alpha):
    # Direction vectors
    v1 = np.array([5, 3 - alpha, -2])
    v2 = np.array([alpha, -1, 2 - alpha])

    # Form matrix with direction vectors as columns
    M = np.column_stack((v1, v2))

    # Rank of coefficient matrix
    rank = np.linalg.matrix_rank(M)

    # Since both lines pass through origin, always coplanar
    return rank <= 2
# Test for some sample alpha values
alpha_values = [-3, -1, 0, 1, 2, 3, 5]
for a in alpha_values:
    result = check_coplanarity(a)
    print(f"alpha = {a:2d} -> Coplanar: {result}")

print("\nConclusion: Lines are coplanar for all real alpha.")

File: ai25btech11007/MatGeo/MatGeo_10/Codes/main.c
#include <stdio.h>

// Function to compute determinant of 3x2 augmented with 0 column
// Actually we compute rank check: det([n1 n2]) = 0 always
int main() {
    int alpha;
    printf("Lines:\n");
    printf("L1: x/5 = y/(3-alpha) = z/(-2)\n");
    printf("L2: x/alpha = y/(-1) = z/(2-alpha)\n\n");

    printf("Checking coplanarity for sample alpha values...\n");

    for(alpha = -3; alpha <= 5; alpha++) {
        int n1[3] = {5, 3 - alpha, -2};
        int n2[3] = {alpha, -1, 2 - alpha};

        // Compute determinant of 3x2 matrix (v1, v2)
        // In coplanarity test: rank <= 2 always (since both pass through origin)
        // Let's check cross product
        int cross[3];
        cross[0] = n1[1]*n2[2] - n1[2]*n2[1];
        cross[1] = n1[2]*n2[0] - n1[0]*n2[2];
        cross[2] = n1[0]*n2[1] - n1[1]*n2[0];

        printf("alpha = %d -> Lines are coplanar (always true)\n", alpha);
    }

    printf("\nConclusion: Lines are coplanar for all real alpha.\n");
    return 0;
}
